spark/src/main/scala/org/apache/spark/sql/delta/util/PartitionUtils.scala

```diff
@@ -528,46 +545,47 @@ private[delta] object PartitionUtils {
       timeZone: TimeZone,
       dateFormatter: DateFormatter,
       timestampFormatter: TimestampFormatter): Literal = {
-    val decimalTry = Try {
-      // `BigDecimal` conversion can fail when the `field` is not a form of number.
-      val bigDecimal = new JBigDecimal(raw)
-      // It reduces the cases for decimals by disallowing values having scale (eg. `1.1`).
-      require(bigDecimal.scale <= 0)
-      // `DecimalType` conversion can fail when
-      //   1. The precision is bigger than 38.
-      //   2. scale is bigger than precision.
-      Literal(bigDecimal)
-    }
 
-    val dateTry = Try {
-      // try and parse the date, if no exception occurs this is a candidate to be resolved as
-      // DateType
-      dateFormatter.parse(raw)
-      // SPARK-23436: Casting the string to date may still return null if a bad Date is provided.
-      // This can happen since DateFormat.parse  may not use the entire text of the given string:
-      // so if there are extra-characters after the date, it returns correctly.
-      // We need to check that we can cast the raw string since we later can use Cast to get
-      // the partition values with the right DataType (see
-      // org.apache.spark.sql.execution.datasources.PartitioningAwareFileIndex.inferPartitioning)
-      val dateValue = Cast(Literal(raw), DateType).eval()
-      // Disallow DateType if the cast returned null
-      require(dateValue != null)
-      Literal.create(dateValue, DateType)
-    }
+    if (typeInference) {
```

Comment:
This is a refactoring to only attempt the attempts to parse into a specific type only if type inference is enabled. I raised it separately: https://github.com/delta-io/delta/pull/3374

after that gets in first, I can rebase so this change is easier to review